demo/item-userlists.ts

```diff
@@ -0,0 +1,244 @@
+      <path d="m56 23v22h22v11h-22v22h-11l-.001-22h-21.999v-11h21.999l.001-22z" />
+    </g>
+  </svg>`;
+  }
+
+  renderUserlistOption(option: userlistOptionInterface): TemplateResult {
+    const { label, isSelected, id } = option;
+    const selected = isSelected ? 'selected' : '';
+    const component = html`<button
+      id="${id}"
+      @click=${() => this.optionClicked(option)}
+    >
+      ${label}
+    </button> `;
+
+    return html`<li class=${selected}>${component}</li>`;
+  }
+
+  optionClicked(option: userlistOptionInterface): void {
+    this.dispatchEvent(
+      new CustomEvent('optionSelected', {
+        detail: { option },
+      })
+    );
+    option.selectedHandler?.(option);
+  }
+
+  private checkedIcon(checked: boolean): TemplateResult {
+    if (checked) {
+      return html`${this.checkIcon}`;
+    }
+    return html`<div style="width: 12px; height: 12px;"></div>`;
+  }
+
+  // Convert userlist data into a list of options
+  get userListOptions(): userlistOptionInterface[] {
+    const options: userlistOptionInterface[] = [];
+
+    this.lists.forEach(list => {
+      const listOption = {
+        label: html` <ia-icon-label>
+          <div slot="icon">${this.checkedIcon(list.item_is_member)}</div>
+          <div class="truncate">${list.name}</div>
+        </ia-icon-label>`,
+        id: list.id,
+        selectedHandler: (option: userlistOptionInterface) =>
+          this.onSelected(option),
+      } as userlistOptionInterface;
+      options.push(listOption);
+    });
+
+    const createNewListOption: userlistOptionInterface = {
+      label: html`<ia-icon-label>
+        <div slot="icon">${this.plusIcon}</div>
+        Create new list
+      </ia-icon-label>`,
+      id: 'create-new-list',
+      selectedHandler: (option: userlistOptionInterface) =>
+        this.onSelected(option),
+    };
+    options.push(createNewListOption);
+
+    return options;
+  }
+
+  private onKeydown(e: KeyboardEvent): void {
+    if (e.key === 'Escape') {
+      this.closeDropdown();
+    }
+  }
```

Comment:
Example of what I mean in the previous comment about arrow function binding:

```suggestion
  // Arrow function to maintain desired `this` binding
  private onKeydown = (e: KeyboardEvent): void => {
    if (e.key === 'Escape') {
      this.closeDropdown();
    }
  }
```